crates/subspace-farmer/src/utils/segment_reconstruction.rs

```diff
@@ -0,0 +1,79 @@
+use std::iter;
+use futures::stream::{FuturesOrdered, FuturesUnordered};
+use futures::StreamExt;
+use subspace_core_primitives::{Piece, PieceIndex, SegmentIndex, PIECES_IN_SEGMENT, RECORD_SIZE};
+use subspace_farmer_components::plotting::PieceGetter;
+use tracing::{error, info};
+use subspace_archiving::reconstructor::Reconstructor;
+
+const SEGMENT_SIZE: u32 = RECORD_SIZE * PIECES_IN_SEGMENT / 2;
```

Comment:
There is already a constant `subspace_core_primitives::RECORDED_HISTORY_SEGMENT_SIZE` for this purpose.